snap/local/runtime-helpers/bin/security-secretstore-setup-launch.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash -ex
+
+security-secretstore-setup -confdir "$SNAP_DATA/config/security-secretstore-setup/res" --vaultInterval=10
+
+if [ -n "$REDIS5_PASSWORD_PATHNAME" ]; then
```

Comment:
When would this be not true?

Reply:
Nice catch, since the env var is always set prior to this script being launched, the length of the string is always non-zero. Instead this should check if the file exists, and the security-secretstore-read command should also get moved into this if block. Here's the diff I'm testing now:
```

diff --git a/snap/local/runtime-helpers/bin/security-secretstore-setup-launch.sh b/snap/local/runtime-helpers/bin/security-secretstore-setup-launch.sh
index 234bd58..5cfc20a 100755
--- a/snap/local/runtime-helpers/bin/security-secretstore-setup-launch.sh
+++ b/snap/local/runtime-helpers/bin/security-secretstore-setup-launch.sh
@@ -2,7 +2,8 @@
 
 security-secretstore-setup -confdir "$SNAP_DATA/config/security-secretstore-setup/res" --vaultInterval=10
 
-if [ -n "$REDIS5_PASSWORD_PATHNAME" ]; then
+# if redis pw file doesn't exist, handle initialization & read it
+if [ ! -f "$REDIS5_PASSWORD_PATHNAME" ]; then
 
     CONFIG_FILE="$SNAP_DATA/config/security-secretstore-read/res/configuration.toml"
     # grep -q returns 0 if a match is found
@@ -18,6 +19,8 @@ if [ -n "$REDIS5_PASSWORD_PATHNAME" ]; then
 
         mv "$CONFIG_FILE.new" "$CONFIG_FILE"
     fi
+
+    # read the pw from secretstore & write to disk
+    security-secretstore-read -confdir "$SNAP_DATA/config/security-secretstore-read/res"
 fi
 
-security-secretstore-read -confdir "$SNAP_DATA/config/security-secretstore-read/res"

```

Reply:
also I have the same question, but for your new patch, under what conditions would security-secretstore-setup-launch.sh get run when this file doesn't exist?  Do we allow users to configure to use redis w/o a password

Reply:
At the moment, security-secretstore-setup-launch.sh is always run in order to launch security-secretstore-setup, which is a one-shot service. The file should not exist when this script is run.  We currently do not allow users to configure mongo or redis without password authentication.

Reply:
This has been addressed in my latest update